experimental/iterators/lib/Dialects/Tabular/IR/Tabular.cpp

```diff
@@ -0,0 +1,124 @@
+//===-- Tabular.cpp - Tabular dialect ---------------------------*- C++ -*-===//
+//
+// Licensed under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#include "iterators/Dialect/Tabular/IR/Tabular.h"
+
+#include "mlir/Dialect/LLVMIR/LLVMTypes.h"
+#include "mlir/IR/DialectImplementation.h"
+#include "mlir/Support/LogicalResult.h"
+#include "llvm/ADT/TypeSwitch.h"
+
+using namespace mlir;
+using namespace mlir::iterators;
+
+//===----------------------------------------------------------------------===//
+// Tabular dialect
+//===----------------------------------------------------------------------===//
+
+#include "iterators/Dialect/Tabular/IR/TabularOpsDialect.cpp.inc"
+
+void TabularDialect::initialize() {
+#define GET_OP_LIST
+  addOperations<
+#include "iterators/Dialect/Tabular/IR/TabularOps.cpp.inc"
+      >();
+  addTypes<
+#define GET_TYPEDEF_LIST
+#include "iterators/Dialect/Tabular/IR/TabularOpsTypes.cpp.inc"
+      >();
+}
+
+//===----------------------------------------------------------------------===//
+// Tabular interfaces
+//===----------------------------------------------------------------------===//
+
+#include "iterators/Dialect/Tabular/IR/TabularOpInterfaces.cpp.inc"
+#include "iterators/Dialect/Tabular/IR/TabularTypeInterfaces.cpp.inc"
+
+//===----------------------------------------------------------------------===//
+// Tabular operations
+//===----------------------------------------------------------------------===//
+
+#define GET_OP_CLASSES
+#include "iterators/Dialect/Tabular/IR/TabularOps.cpp.inc"
+
+LogicalResult CreateTabularViewOp::verify() {
+  auto viewType = view().getType().cast<TabularViewType>();
+  TypeRange columnTypes = viewType.getColumnTypes();
+
+  // Verify matching column/element types.
+  if (!llvm::all_of_zip(memrefs().getTypes(), columnTypes,
+                        [](Type t1, Type t2) {
+                          return t1.cast<MemRefType>().getElementType() == t2;
+                        })) {
+
+    std::string viewColumnTypes;
+    {
+      llvm::raw_string_ostream stream(viewColumnTypes);
+      llvm::interleaveComma(columnTypes, stream,
+                            [&](Type type) { type.print(stream); });
+    }
+    std::string memrefColumnTypes;
+    {
+      llvm::raw_string_ostream stream(memrefColumnTypes);
+      llvm::interleaveComma(memrefs().getTypes(), stream, [&](Type type) {
+        type.cast<MemRefType>().getElementType().print(stream);
+      });
+    }
+    return emitOpError()
```

Comment:
I would just iterate with an index and report that `element type of operand "index" (i.e. memref<...>) does not match the result "index" column type (i.e. resulttype<...>))`.
The message is both more actionnable and the code quite simpler as you just use vanilla type printing and avoid managing your strings.